desc/objectives/linear_objectives.py

```diff
@@ -275,6 +275,7 @@ def __init__(
             name=name,
         )
 
+    @set_default_cpu
```

Comment:
Do we want to apply this to all the build methods? It seems kind of inconsistent with where it is used now.

Reply:
I've added it to the parent objective class before and for my test case it didn't give better results, so I picked suitable build methods intuitively. 

Also, maybe one important thing to keep in mind is when we build smt on cpu, probably the arrays are on cpu memory, we may need to copy them to gpu memory as we did during hackathon.